DopplerCurrencyJob/DopplerCurrencyService/DopplerCurrencyService.cs

```diff
@@ -173,5 +171,35 @@ private string CreateJwtToken()
 
             return _tokenHandler.WriteToken(jwtToken);
         }
+
+        private async Task<CurrencyResponse> GetPreviousPrices(DateTime cstTime, string currencyCode)
+        {
+            var currentDate = cstTime;
+            CurrencyResponse result = null;
+            var count = 0;
+
+            while (count < _dopplerCurrencySettings.HolidayRetryCountLimit)
+            {
+                cstTime = cstTime.DayOfWeek == DayOfWeek.Monday ? cstTime.AddDays(-3) : cstTime.AddDays(-1);
+                var httpResponse = await GetCurrencyValue(currencyCode, cstTime);
```

Comment:
I do not sure create a new token per each retry, maybe we can use the only token per each retry

Reply:
Yes, Agree, I will update it.